src/core/server/saved_objects/permission_control/acl.ts

```diff
@@ -0,0 +1,334 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export enum PrincipalType {
+  Users = 'users',
+  Groups = 'groups',
+}
+
+export interface Principals {
+  users?: string[];
+  groups?: string[];
+}
+
+export type Permissions = Record<string, Principals>;
+
+export interface TransformedPermission {
+  type: string;
+  name: string;
+  permissions: string[];
+}
+
+const addToPrincipals = ({
+  principals = {},
+  users,
+  groups,
+}: {
+  principals: Principals;
+  users?: string[];
+  groups?: string[];
+}) => {
+  if (!!users) {
```

Comment:
is it necessary to use double exclamation (!!) ?

Reply:
No, it's not necessary, I'll change it, thanks!